GoodreadsEDA/bk_plt.py

```diff
@@ -1,98 +1,131 @@
-                                   WHERE book_title LIKE "Billy Lynn%"
-                                   ;""", conn)
-        else:
-            df = pd.read_sql_query(f"""SELECT * FROM Reviews WHERE book_title LIKE "{bk}%";""", conn)
-
-        conn.close()
-
-        df.drop(["matching_title", "book_url"], axis=1, inplace=True)
-
-        # turn string dates into datetime objects, and sort the dataframe by date.
-        df["review_date"] = pd.to_datetime(df["review_date"])
-        df.set_index("review_date", inplace=True)
-        df.sort_index(inplace=True)
-
-        # change review scores to ints, if it cannot be converted to int, NaN.
-        df["review_score"] = df["review_score"].apply(pd.to_numeric, errors='coerce')
-
-        # drop rows with NaN values in them.
-        df.dropna(inplace=True)
-
-        # take rolling average of every fifteen review scores.
-        df["review_score_rolling"] = df["review_score"].rolling(window=n_reviews,
-                                                                center=False).mean()
-
-        # plotting parameters
-        x = df.index
-        y = df["review_score_rolling"]
-
-        plt.plot(x, y)
-
-        plt.yticks(np.arange(2.5, 5.5, .5)) # books are seldom given a score < 3.
-        plt.ylabel(f"Average Rating Per {n_reviews} Reviews")
-
-        # plot release date of movie as a line and label it.
-        plt.axvline(x=mv_release, color="#f99f75")
-        mv_release_datetime = datetime.strptime(mv_release, "%d %B %Y")
-        plt.text(mv_release_datetime + timedelta(days=10), 4,
-                                                 'Movie Release Date',
-                                                 rotation=90,
-                                                 color='#767a78')
-
-        plt.xlabel("Year")
-        plt.xticks(rotation=45)
-        plt.xlim(mv_release_datetime - relativedelta(years=1),
-                 mv_release_datetime + relativedelta(years=1))
-        plt.title(f"Review Scores of '{bk}' over time.")
-        plt.subplots_adjust(bottom=0.2)
-        sns.despine()
-        plt.show()
-    except ValueError as e:
-        print(e)
-        print(df)
-        continue
+
+def plot_results(df, n_reviews, book_title, mv_release):
+    """Plots the relationship between movie release and book ratings over time.
+
+    Args:
+        df (pandas.DataFrame): dataframe containing the values to plot.
+        n_reviews (int): denominator to use for calculating rolling average.
+        book_title (str): title of the book to plot.
+        mv_release (str): the release date of the movie of the book.
+    """
+
+    # getting movie release date
+    mv_release_datetime = datetime.strptime(mv_release, "%d %B %Y")
+    # setting plotting parameters
+    set_plotting_params()
+
+    # plotting rolling average of book reviews over time and date of movie
+    # release
+    df.plot(x=df.index, y='review_score_rolling', color="#f99f75")
```

Comment:
It's certainly not the best for a lot of uses, but for just plotting a line, I think this is the function you wanna go for. There's also space here to put every plot on one graph, or maybe the books whose ratings increased the most from their movie release or something.

Reply:
Yeah, what this script outputs certainly needs some drawing board time. If I'm to dive deeper I was considering plotting the average review of the rest of the author's books on the same time series to see if the average rating of their book that was given an adaptation deviated from there other books in any meaningful way. Once the code base is a little more clear of cobwebs I'll evaluate my options w/r/t this.